File: elixir/pangram/pangram.exs
defmodule Pangram do

  @alphabet Enum.map(?a..?z, &<<&1::utf8>>)

  @spec pangram?(String.t()) :: boolean
  def pangram?(""), do: false
  def pangram?(sentence) do
    pangram =
      sentence
      |> String.codepoints()
      |> Enum.reduce([], fn letter, pangram ->
        letter = String.downcase(letter)
        if letter in @alphabet and letter not in pangram, do: [letter | pangram], else: pangram
      end)
      |> Enum.sort()

    pangram === @alphabet
  end
end

File: ruby/pangram/pangram.rb
class Pangram
  def self.pangram?(sentence)
    alphabet = ('a'..'z').to_a
    pangram = []

    sentence.chars.each do |char|
      char = char.downcase
      if alphabet.include?(char) and not pangram.include?(char)
        pangram.push(char)
      end
    end

    pangram.sort == alphabet
  end
end
